src/olympia/abuse/models.py

```diff
@@ -316,16 +331,34 @@ def delete(self, *args, **kwargs):
         # soft-deleted.
         return self.update(state=self.STATES.DELETED)
 
+    def save(self, *args, **kwargs):
+        from olympia.abuse.tasks import report_to_cinder
+
+        creation = kwargs.get('force_insert', False) or bool(self.pk)
+        rval = super().save(*args, **kwargs)
+        if (
+            creation
+            and waffle.switch_is_active('enable-cinder-reporting')
+            and self.reason in AbuseReport.REASONS.REPORTABLE_REASONS
+        ):
+            # call task to fire off cinder report
+            report_to_cinder.delay(self.pk)
+        return rval
```

Comment:
I'm not sure about this change - it'll be reporting to cinder automatically ever time an instance is created which seems extreme (even in tests!).  What was the problem in firing the task in the serializer?

Reply:
At the minimum I needed to move that out of the addon-specific serializer, so I thought I'd do this at the model level while I'm at it, which is cleaner - ultimately we shouldn't care that it's an API request, we have to do this for every abuse report.

I should address tests, but also we are likely going to need something along those lines when we implement https://mozilla-hub.atlassian.net/browse/AMO-116 (though I'm not sure we'll have `AbuseReport` instances for that)

Reply:
IMO the model save just seems too low level to trigger a request to a remote service that creates new records.  If we in the future (for whatever reason) create an AbuseReport instance to backfill anything, or in the admin tools that would also trigger a (hidden) cinder request.

https://mozilla-hub.atlassian.net/browse/AMO-116 I expect would need special handling, because we're going to want to immediately resolve the job after it's created.

Reply:
we could move all the logic to a utility function if that's cleaner?

Reply:
I'll go back to a serializer implementation for now, I don't think a utility function helps much.